compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10283,14 +10283,14 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
             {
             if (return102)
                {
-               generateRIInstruction(cg, TR::InstOpCode::LA, node, retValReg, 2);
+               generateRIInstruction(cg, TR::InstOpCode::LGFI, node, retValReg, 2);
```

Comment:
Similar to https://github.com/eclipse/omr/pull/3127#discussion_r228721789 we should use `TR::InstOpCode::getLoadHalfWordImmOpCode()` to use LGHI or LHI. https://github.com/eclipse/omr/blob/a44b67653214e2884d9890937b08e65e60d3d308/compiler/z/codegen/InstOpCode.cpp#L274